File: Airship/AirshipCore/Source/common/UAAppExitEvent.m
/* Copyright Airship and Contributors */

#import "UAAppExitEvent+Internal.h"
#import "UAEvent+Internal.h"
#import "UAAnalytics.h"
#import "UAirship.h"
#import "UAUtils+Internal.h"

@implementation UAAppExitEvent

+ (instancetype)event {
    NSMutableDictionary *data = [NSMutableDictionary dictionary];
    [data setValue:[UAirship analytics].conversionSendID forKey:@"push_id"];
    [data setValue:[UAirship analytics].conversionPushMetadata forKey:@"metadata"];
    [data setValue:[UAUtils connectionType] forKey:@"connection_type"];

    UAAppExitEvent *event = [[self alloc] init];
    event.data = [data copy];
    return event;
}

- (NSString *)eventType {
    return @"app_exit";
}

@end

File: Airship/AirshipCore/Source/UAAppExitEvent.swift
/* Copyright Airship and Contributors */

/**
 * @note For Interrnal use only :nodoc:
 */
@objc
public class UAAppExitEvent : NSObject, UAEvent {
    private let _data : [AnyHashable : Any]

    @objc
    public var priority: UAEventPriority {
        get {
            return .normal
        }
    }

    @objc
    public var eventType : String {
        get {
            return "app_exit"
        }
    }

    @objc
    public var data: [AnyHashable : Any] {
        get {
            return self._data
        }
    }

    @objc
    public override init() {
        let analytics = Airship.requireComponent(ofType: AnalyticsProtocol.self)
        var data: [AnyHashable : Any] = [:]
        data["push_id"] = analytics.conversionSendID
        data["metadata"] = analytics.conversionPushMetadata
        data["connection_type"] = UAUtils.connectionType()
        self._data = data

        super.init()
    }
}
